metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/MetadataImporterMainApplication.java

```diff
@@ -45,22 +44,23 @@
 
     private ObjectsImporter objectsImporter;
 
-    private StudyRepository studyRepository;
-
-    public MetadataImporterMainApplication(ObjectsImporter objectsImporter, StudyRepository studyRepository) {
+    public MetadataImporterMainApplication(ObjectsImporter objectsImporter) {
         this.objectsImporter = objectsImporter;
-        this.studyRepository = studyRepository;
     }
 
     public static void main(String[] args) throws Exception {
         SpringApplication.run(MetadataImporterMainApplication.class, args);
+
     }
 
     @Override
     public void run(ApplicationArguments applicationArguments) throws Exception {
         Set<String> accessions = readAccessionsFromFile(applicationArguments);
-        List<Study> studies = objectsImporter.importStudy(accessions);
-        studyRepository.save(studies);
+        if (objectsImporter instanceof SraObjectsImporterThroughDatabase) {
+            accessions.parallelStream().forEach(accession -> objectsImporter.importAnalysis(accession));
+        } else {
```

Comment:
Check the appropriate `instanceof` here too for complete correctness, and add an `else` branch that throws an exception otherwise.